src/components/page/FoodsPage.tsx

```diff
@@ -1,3 +1,156 @@
-export const FoodsPage = () => {
-  return <>FoodsPage</>;
+'use client';
+
+import { ContainerIcon, HomeIcon_Off, MenuMeatballIcon } from '@/assets/images/icons';
+import { Icon } from '@/components/ui';
+import { foodCategories } from '@/const/foodCategory';
+import { FilterBadge } from '@/features/foods/components';
+import { FilterButton } from '@/features/foods/components/FilterButton';
+import { FoodList } from '@/features/foods/components/FoodList';
+import { FoodSort } from '@/features/foods/components/FoodSort';
+import { SearchBar } from '@/features/foods/components/SearchBar';
+import { IFoodView } from '@/features/foods/types/utils';
+import {
+  createContainerIdNameMap,
+  createGroupIdNameMap,
+  groupContainersByGroupId,
+} from '@/features/foods/utils/containerMapping';
+import { IContainer, IFood } from '@/types/definition';
+
+import { Route } from 'next';
+import { usePathname, useRouter, useSearchParams } from 'next/navigation';
+import React, { useEffect, useState } from 'react';
+
+export const FoodsPage = ({ containers }: { containers: IContainer[] }) => {
+  const pathname = usePathname();
+  const searchParams = useSearchParams();
+  const params = new URLSearchParams(searchParams);
+  const { replace } = useRouter();
+  const sort = searchParams?.get('sort') || '';
+  const group = searchParams?.get('group') || '';
+  const container = searchParams?.get('container') || '';
+  const [displayedFoods, setDisplayedFoods] = useState<IFoodView[]>([]);
+  const [query, setQuery] = useState<string>(searchParams?.get('query') || '');
+  const [categoryList, setCategoryList] = useState<string[]>(
+    searchParams?.get('category')?.split(',') || [],
+  );
+
+  const containersGroupByGroups = groupContainersByGroupId(containers);
```

Comment:
Thank you for your suggestion!!
I agree with your opinion to be more understandable by setting `GroupId` instead of `Groups`.
So I would like to choose this.

- `containersGroupByGroupId`

> It should be at least "grouped", not "group"

In SQL grammar, we use `GROUP BY` so I thought we can easily grasp this meaning with it. and it causes to make the variable longer. How do you think?
https://www.w3schools.com/sql/sql_groupby.asp#:~:text=The%20SQL%20GROUP%20BY%20Statement,by%20one%20or%20more%20columns.